File: typescript/session3/MotRecursif/MotNonVideInductif.ts

import { Char } from "../Char";
import { Mot } from "./mot";

export class MotNonVideInductif extends Mot {

  private first: Char;
  private suiteMot: Mot;

  public constructor(first: Char, suite: Mot) {
    super();
    this.first = first;
    this.suiteMot = suite;
  }

  public taille(): number {
    return 1 + this.suiteMot.taille();
  }

  public estCons(): boolean {
    return true;
  }

  public premiereLettre(): Char {
    return this.first;
  }

  public suite(): Mot {
    return this.suiteMot;
  }

  public getValue(): string {
    return this.premiereLettre.toString() + this.suiteMot.getValue();
  }

}

File: java/cours/src/session3/TD/MotRecursif/MotNonVideInductif.java
package session3.TD.MotRecursif;

public class MotNonVideInductif implements Mot {
	private char first;
	private Mot suite;

	public MotNonVideInductif(char first, Mot suite) {
		this.first = first;
		this.suite = suite;
	}

	@Override
	public int taille() {
		return 1 + suite.taille();
	}

	@Override
	public boolean estConcat() {
		return true;
	}

	@Override
	public char premiereLettre() {
		return this.first;
	}

	@Override
	public Mot suite() {
		return this.suite;
	}

	@Override
	public String getValue() {
		return this.first + this.suite.getValue();
	}
}
